core/src/main/java/com/sequenceiq/cloudbreak/job/existingstackfix/ExistingStackPatcherJob.java

```diff
@@ -0,0 +1,112 @@
+package com.sequenceiq.cloudbreak.job.existingstackfix;
+
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.StackPatchType;
+import com.sequenceiq.cloudbreak.domain.view.StackView;
+import com.sequenceiq.cloudbreak.quartz.statuschecker.job.StatusCheckerJob;
+import com.sequenceiq.cloudbreak.service.stackpatch.ExistingStackPatchApplyException;
+import com.sequenceiq.cloudbreak.service.stackpatch.ExistingStackPatchService;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.cloudbreak.service.stack.StackViewService;
+
+import io.opentracing.Tracer;
+
+@DisallowConcurrentExecution
+@Component
+public class ExistingStackPatcherJob extends StatusCheckerJob {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExistingStackPatcherJob.class);
+
+    private static final EnumSet<Status> FAILED_OR_IN_DELETE_STATUSES = EnumSet.of(
+            Status.CREATE_FAILED,
+            Status.PRE_DELETE_IN_PROGRESS,
+            Status.DELETE_IN_PROGRESS,
+            Status.DELETE_FAILED,
+            Status.DELETE_COMPLETED,
+            Status.EXTERNAL_DATABASE_CREATION_FAILED,
+            Status.EXTERNAL_DATABASE_DELETION_IN_PROGRESS,
+            Status.EXTERNAL_DATABASE_DELETION_FINISHED,
+            Status.EXTERNAL_DATABASE_DELETION_FAILED,
+            Status.LOAD_BALANCER_UPDATE_FAILED
+    );
+
+    @Inject
+    private StackViewService stackViewService;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private ExistingStackPatcherJobService jobService;
+
+    @Inject
+    private Collection<ExistingStackPatchService> existingStackPatchServices;
+
+    public ExistingStackPatcherJob(Tracer tracer) {
+        super(tracer, "Existing Stack Patcher Job");
+    }
+
+    @Override
+    protected Object getMdcContextObject() {
+        return stackViewService.findById(getStackId()).orElseGet(StackView::new);
+    }
+
+    @Override
+    protected void executeTracedJob(JobExecutionContext context) throws JobExecutionException {
+        Stack stack = stackService.getByIdWithListsInTransaction(getStackId());
+        Status stackStatus = stack.getStatus();
+        if (!FAILED_OR_IN_DELETE_STATUSES.contains(stackStatus)) {
```

Comment:
This would rather be active or running clusters right?

Reply:
What if there is an ongoing Scale or Upgrade operation? Shouldn't there be a check for no Flow running for an operation? How long does the stack patch takes?